### PR TITLE
OCPBUGS-23386: Generate FIPS compatible RHEL9 oc binary

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR generates RHEL9 FIPS enabled compatible oc binaries in linux_amd64,
linux_arm64, linux_ppc64le architectures and saves them as oc.rhel9(this will be backported to 4.15 and 4.14).

### CODE REVIEW

**Comment by ardaguclu** [comment]
But first we need a separate PR in https://github.com/openshift-eng/ocp-build-data repository.

/hold

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-8-release-golang-1.21-openshift-4.16
+  tag: rhel-8-release-golang-1.20-openshift-4.16
```
Comment (soltysh) [inline .ci-operator.yaml:4]: Why the change?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/openshift/oc
 
-go 1.21
+go 1.20
```
Comment (soltysh) [inline go.mod:3]: Same, I'd stick with 1.21

File: images/cli/Dockerfile.rhel [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.21-openshift-4.16 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.16 AS builder
```
Comment (soltysh) [inline images/cli/Dockerfile.rhel:1]: Also here

**Review by soltysh — APPROVED** [review]
/lgtm
